optuna/integration/_lightgbm_tuner/optimize.py

```diff
@@ -728,18 +729,17 @@ def __init__(
         if valid_sets is None:
             raise ValueError("`valid_sets` is required.")
 
-    @property
+    @property  # type: ignore
+    @deprecated(
+        "1.4.0",
+        "3.0.0",
+        text=(
+            "Please get the best booster via "
+            ":class:`~optuna.integration.lightgbm.LightGBMTuner.get_best_booster` instead."
+        ),
+    )
```

Comment:
Mostly wondering but does having the `deprecated` also work?

Reply:
I'm sorry but I couldn't get the point. If you mean the confirmation of the behavior of `best_booster` property with `@decorated`, it works as expected :

```python
# examples/lightgbm_tuner_simple.py

    ....
    # model = lgb.train(
    #     params, dtrain, valid_sets=[dtrain, dval], verbose_eval=100, early_stopping_rounds=100,
    # )
    # model = lgb.train(
    #     params, dtrain, valid_sets=[dtrain, dval], verbose_eval=100, early_stopping_rounds=100,
     # )
    lgbt = lgb.LightGBMTuner(
        params, dtrain, valid_sets=[dtrain, dval], verbose_eval=100, early_stopping_rounds=100,
    )
    lgbt.run()
    model = lgbt.best_booster
```

```console
optuna/_deprecated.py:110: DeprecationWarning: best_booster has been deprecated in v1.4.0. This feature will be removed in v3.0.0.
  DeprecationWarning,
```

Reply:
Sorry, I meant having the `deprecated` decorator outermost (applied after `property`).